internal/k8s/controller.go

```diff
@@ -1400,6 +1400,73 @@ func (lbc *LoadBalancerController) emitEventForVirtualServers(eventType string,
 	}
 }
 
+func getStatusFromEventTitle(eventTitle string) string {
+	switch eventTitle {
+	case "AddedOrUpdatedWithError", "Rejected", "NoVirtualServersFound", "Missing Secret", "UpdatedWithError":
+		return conf_v1.StateInvalid
+	case "AddedOrUpdatedWithWarning", "UpdatedWithWarning":
+		return conf_v1.StateWarning
+	case "AddedOrUpdated", "Updated":
+		return conf_v1.StateValid
+	default:
+		return "Unknown"
+	}
+}
+
+func (lbc *LoadBalancerController) updateVirtualServerVirtualServerRouteStatusesFromEvents() error {
```

Comment:
optional: add and `and` between VirtualServer and VirtualServerRoutes if it doesn't make the name too long